src/Algebra/Graph/Acyclic/Ord.hs

```diff
@@ -0,0 +1,164 @@
+-----------------------------------------------------------------
+-- |
+-- Module     : Algebra.Graph..Acyclic.Ord
+-- License    : MIT (see the file LICENSE)
+-- Stability  : experimental
+--
+-- __Alga__ is a library for algebraic construction and manipulation of graphs
+-- in Haskell. See <https://github.com/snowleopard/alga-paper this paper> for the
+-- motivation behind the library, the underlying theory, and implementation details.
+--
+-- This module defines functions which guarantee Acyclic graph by
+-- establishing an invariant on the edges.
+-- An edge (x, y) exists if and only if x < y. This way no back
+-- edges are formed and hence making a cyclic graph is impossible.
+----------------------------------------------------------------
+module Algebra.Graph.Acyclic.Ord (
+    -- * Graph construction primitives
+    edge, overlay, connect, edges, overlays, connects,
+
+    -- * Graph transformation
+    replaceVertex, mergeVertices, gmap,
+    ) where
+
+import Algebra.Graph.Acyclic.AdjacencyMap
+import qualified Algebra.Graph.AdjacencyMap as AM
+
+-- | Construct the graph comprising /a single edge/.
+-- Complexity: /O(1)/ time, memory.
+--
+-- @
+-- edge x y               == 'connect' ('vertex' x) ('vertex' y)
+-- 'hasEdge' 1 2 (edge 1 2) == True
+-- 'hasEdge' 2 1 (edge 2 1) == False
+-- 'edgeCount'   (edge 1 2) == 1
+-- 'edgeCount'   (edge 2 1) == 0
+-- 'vertexCount' (edge 1 1) == 1
+-- 'vertexCount' (edge 1 2) == 2
+-- 'vertexCount' (edge 2 1) == 2
+-- @
+edge :: Ord a => a -> a -> AdjacencyMap a
+edge x y = toAcyclicOrd (AM.edge x y)
+
+-- | /Overlay/ two graphs. This is a commutative, associative and idempotent
+-- operation with the identity 'empty'.
+-- Complexity: /O((n + m) * log(n))/ time and /O(n + m)/ memory.
+--
+-- @
+-- 'isEmpty'     (overlay x y) == 'isEmpty'   x   && 'isEmpty'   y
+-- 'hasVertex' z (overlay x y) == 'hasVertex' z x || 'hasVertex' z y
+-- 'vertexCount' (overlay x y) >= 'vertexCount' x
+-- 'vertexCount' (overlay x y) <= 'vertexCount' x + 'vertexCount' y
+-- 'edgeCount'   (overlay x y) >= 'edgeCount' x
+-- 'edgeCount'   (overlay x y) <= 'edgeCount' x   + 'edgeCount' y
+-- 'vertexCount' (overlay 1 2) == 2
+-- 'edgeCount'   (overlay 1 2) == 0
+-- @
+overlay :: Ord a => AdjacencyMap a -> AdjacencyMap a -> AdjacencyMap a
+overlay x y = toAcyclicOrd $ AM.overlay (fromAcyclic x) (fromAcyclic y)
```

Comment:
This is only cosmetic, and I don't know what @snowleopard thinks of it, but there is not-so-known high-order function called [`on`](http://hackage.haskell.org/package/base-4.12.0.0/docs/Data-Function.html#v:on) in `base`, of type `(b -> b -> c) -> (a -> b) -> a -> a -> c `, which can maybe be used here to simplify the expression.

Reply:
> not-so-known 

Agreed! I was searching for something like this. Will change it!